File: my-folder/1528-kids-with-the-greatest-number-of-candies/solution.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
       	List<Boolean> canBeGreatest = new ArrayList<Boolean>(candies.length);
		int max = candies[0];
		for (int i = 1; i < candies.length; i++) {
			if (candies[i] > max) {
				max = candies[i];
			}
		}
		for (int i : candies) {
			if (i + extraCandies >= max) {
				canBeGreatest.add(true);
			} else {
				canBeGreatest.add(false);
			}
		}
		return canBeGreatest;
    }
}

File: my-folder/1528-kids-with-the-greatest-number-of-candies/solution.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int maxC = candies[0];
        vector<bool> result;
        for (int i = 1; i < candies.size(); i++)
        {
            if (maxC < candies[i])
            {
                maxC = candies[i];
            }
        }
        for (int i = 0; i < candies.size(); i++)
        {
            result.push_back(candies[i] + extraCandies >= maxC);
        }
        return result;
    }
};
